viadot/flows/bigquery_to_adls.py

```diff
@@ -0,0 +1,172 @@
+from pathlib import Path
+from typing import Any, Dict, List
+import os
+import pendulum
+from prefect import Flow
+from prefect.backend import set_key_value
+from prefect.utilities import logging
+
+logger = logging.get_logger()
+
+from ..task_utils import (
+    df_get_data_types_task,
+    add_ingestion_metadata_task,
+    df_to_csv,
+    df_to_parquet,
+    dtypes_to_json_task,
+    df_map_mixed_dtypes_for_parquet,
+    update_dtypes_dict,
+)
+
+from ..tasks import BigQueryToDF
+from ..tasks import AzureDataLakeUpload
+
+bigquery_to_df_task = BigQueryToDF()
+file_to_adls_task = AzureDataLakeUpload()
+json_to_adls_task = AzureDataLakeUpload()
+
+logger = logging.get_logger(__name__)
+
+
+class BigQueryToADLS(Flow):
+    def __init__(
+        self,
+        name: str = None,
+        dataset: str = None,
+        table: str = None,
+        date_column_name: str = "date",
+        start_date: str = None,
+        end_date: str = None,
+        credentials_key: str = "BIGQUERY",
+        output_file_extension: str = ".parquet",
+        adls_dir_path: str = None,
+        local_file_path: str = None,
+        adls_file_name: str = None,
+        adls_sp_credentials_secret: str = None,
+        if_exists: str = "replace",
+        *args: List[any],
+        **kwargs: Dict[str, Any],
+    ):
+        """
```

Comment:
maybe it will be worth to put also here the info about date ranges ( these 3 scenarios yesterday etc)